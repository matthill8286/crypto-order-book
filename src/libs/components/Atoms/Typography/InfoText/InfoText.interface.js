
//# sourceMappingURL=InfoText.interface.js.map