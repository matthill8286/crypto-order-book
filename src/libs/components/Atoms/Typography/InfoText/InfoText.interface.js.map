{"version":3,"sources":[],"names":[],"mappings":"","sourcesContent":[],"file":"InfoText.interface.js"}