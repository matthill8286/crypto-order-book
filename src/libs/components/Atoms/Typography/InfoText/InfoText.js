function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import { Typo } from "../Typo";
export var InfoText = function InfoText(_ref) {
  var _ref$tag = _ref.tag,
      tag = _ref$tag === void 0 ? 'span' : _ref$tag,
      _ref$scale = _ref.scale,
      scale = _ref$scale === void 0 ? 'small' : _ref$scale,
      _ref$color = _ref.color,
      color = _ref$color === void 0 ? 'grey4' : _ref$color,
      weight = _ref.weight,
      fontSize = _ref.fontSize,
      children = _ref.children,
      restProps = _objectWithoutProperties(_ref, ["tag", "scale", "color", "weight", "fontSize", "children"]);

  var isSmall = scale === 'small';
  var isExtraSmall = scale === 'extra-small';
  var isHighlighted = scale === 'small-highlighted';
  var presetFontSize = (isSmall || isHighlighted) && 'xxs' || isExtraSmall && 'xxxs';
  var presetFontWeight = (isSmall || isExtraSmall) && 'regular' || isHighlighted && 'semibold';
  return React.createElement(Typo, _extends({
    tag: tag,
    color: color,
    fontSize: fontSize || presetFontSize || undefined,
    weight: weight || presetFontWeight || undefined
  }, restProps), children);
};
InfoText.displayName = 'InfoText';
//# sourceMappingURL=InfoText.js.map