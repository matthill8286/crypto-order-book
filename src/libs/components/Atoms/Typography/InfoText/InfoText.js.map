{"version":3,"sources":["../../../../../src/components/Atoms/Typography/InfoText/InfoText.tsx"],"names":["React","Typo","InfoText","tag","scale","color","weight","fontSize","children","restProps","isSmall","isExtraSmall","isHighlighted","presetFontSize","presetFontWeight","undefined","displayName"],"mappings":";;;;;;AAAA,OAAOA,KAAP,MAAkB,OAAlB;AACA,SAASC,IAAT;AAGA,OAAO,IAAMC,QAAiC,GAAG,SAApCA,QAAoC,OAQ3C;AAAA,sBAPJC,GAOI;AAAA,MAPJA,GAOI,yBAPE,MAOF;AAAA,wBANJC,KAMI;AAAA,MANJA,KAMI,2BANI,OAMJ;AAAA,wBALJC,KAKI;AAAA,MALJA,KAKI,2BALI,OAKJ;AAAA,MAJJC,MAII,QAJJA,MAII;AAAA,MAHJC,QAGI,QAHJA,QAGI;AAAA,MAFJC,QAEI,QAFJA,QAEI;AAAA,MADDC,SACC;;AACJ,MAAMC,OAAO,GAAGN,KAAK,KAAK,OAA1B;AACA,MAAMO,YAAY,GAAGP,KAAK,KAAK,aAA/B;AACA,MAAMQ,aAAa,GAAGR,KAAK,KAAK,mBAAhC;AAEA,MAAMS,cAAc,GAAI,CAACH,OAAO,IAAIE,aAAZ,KAA8B,KAA/B,IAA0CD,YAAY,IAAI,MAAjF;AACA,MAAMG,gBAAgB,GAAI,CAACJ,OAAO,IAAIC,YAAZ,KAA6B,SAA9B,IAA6CC,aAAa,IAAI,UAAvF;AAEA,SACE,oBAAC,IAAD;AACE,IAAA,GAAG,EAAET,GADP;AAEE,IAAA,KAAK,EAAEE,KAFT;AAGE,IAAA,QAAQ,EAAEE,QAAQ,IAAIM,cAAZ,IAA8BE,SAH1C;AAIE,IAAA,MAAM,EAAET,MAAM,IAAIQ,gBAAV,IAA8BC;AAJxC,KAKMN,SALN,GAMGD,QANH,CADF;AAUD,CA1BM;AA4BPN,QAAQ,CAACc,WAAT,GAAuB,UAAvB","sourcesContent":["import React from 'react'\nimport { Typo } from '../Typo'\nimport { InfoTextProps } from './InfoText.interface'\n\nexport const InfoText: React.FC<InfoTextProps> = ({\n  tag = 'span',\n  scale = 'small',\n  color = 'grey4',\n  weight,\n  fontSize,\n  children,\n  ...restProps\n}) => {\n  const isSmall = scale === 'small'\n  const isExtraSmall = scale === 'extra-small'\n  const isHighlighted = scale === 'small-highlighted'\n\n  const presetFontSize = ((isSmall || isHighlighted) && 'xxs') || (isExtraSmall && 'xxxs')\n  const presetFontWeight = ((isSmall || isExtraSmall) && 'regular') || (isHighlighted && 'semibold')\n\n  return (\n    <Typo\n      tag={tag}\n      color={color}\n      fontSize={fontSize || presetFontSize || undefined}\n      weight={weight || presetFontWeight || undefined}\n      {...restProps}>\n      {children}\n    </Typo>\n  )\n}\n\nInfoText.displayName = 'InfoText'\n"],"file":"InfoText.js"}