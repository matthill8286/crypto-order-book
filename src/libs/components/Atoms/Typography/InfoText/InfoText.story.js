function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { boolean, number, select, text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import * as React from 'react';
import { saiyanTheme } from "../../../../styles/sc-vars-saiyan";
import { InfoText } from '../index';
storiesOf('Design System/Atoms/Typography', module).add('InfoText', function () {
  var sampleText = text('Example text', "Lorem ipsum dolor");
  var scaleOptions = ['small', 'small-highlighted', 'extra-small'];
  var scale = select('Type scale', scaleOptions, 'small');
  var tagOptions = ['span', 'em', 'strong', 'div'];
  var tagName = select('Tag name', tagOptions, 'span');
  var showColors = boolean('Enable Custom Color', false);
  var selectedColor = select('Color', Object.keys(saiyanTheme.color), 'black');
  var limitLines = number('Limit Lines', 0);
  return React.createElement(InfoText, _extends({
    tag: tagName,
    scale: scale
  }, showColors && {
    color: selectedColor
  }, {
    limitLines: limitLines
  }), sampleText);
});
//# sourceMappingURL=InfoText.story.js.map
