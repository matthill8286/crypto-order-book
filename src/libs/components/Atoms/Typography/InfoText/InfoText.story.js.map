{"version":3,"sources":["../../../../../src/components/Atoms/Typography/InfoText/InfoText.story.tsx"],"names":["boolean","number","select","text","storiesOf","React","saiyanTheme","InfoText","module","add","sampleText","scaleOptions","scale","tagOptions","tagName","showColors","selectedColor","Object","keys","color","limitLines"],"mappings":";;AAAA,SAASA,OAAT,EAAkBC,MAAlB,EAA0BC,MAA1B,EAAkCC,IAAlC,QAA8C,wBAA9C;AACA,SAASC,SAAT,QAA0B,kBAA1B;AACA,OAAO,KAAKC,KAAZ,MAAuB,OAAvB;AACA,SAASC,WAAT;AAEA,SAASC,QAAT,QAAyB,IAAzB;AAGAH,SAAS,CAAC,gCAAD,EAAmCI,MAAnC,CAAT,CAAoDC,GAApD,CAAwD,UAAxD,EAAoE,YAAM;AACxE,MAAMC,UAAU,GAAGP,IAAI,CAAC,cAAD,sBAAvB;AACA,MAAMQ,YAAY,GAAG,CAAC,OAAD,EAAU,mBAAV,EAA+B,aAA/B,CAArB;AACA,MAAMC,KAAK,GAAGV,MAAM,CAAC,YAAD,EAAeS,YAAf,EAA6B,OAA7B,CAApB;AACA,MAAME,UAA2C,GAAG,CAAC,MAAD,EAAS,IAAT,EAAe,QAAf,EAAyB,KAAzB,CAApD;AACA,MAAMC,OAAO,GAAGZ,MAAM,CAAC,UAAD,EAAaW,UAAb,EAAyB,MAAzB,CAAtB;AACA,MAAME,UAAU,GAAGf,OAAO,CAAC,qBAAD,EAAwB,KAAxB,CAA1B;AACA,MAAMgB,aAAa,GAAGd,MAAM,CAAC,OAAD,EAAUe,MAAM,CAACC,IAAP,CAAYZ,WAAW,CAACa,KAAxB,CAAV,EAA0C,OAA1C,CAA5B;AACA,MAAMC,UAAU,GAAGnB,MAAM,CAAC,aAAD,EAAgB,CAAhB,CAAzB;AAEA,SACE,oBAAC,QAAD;AACE,IAAA,GAAG,EAAEa,OADP;AAEE,IAAA,KAAK,EAAEF;AAFT,KAGOG,UAAU,IAAI;AAAEI,IAAAA,KAAK,EAAEH;AAAT,GAHrB;AAIE,IAAA,UAAU,EAAEI;AAJd,MAKGV,UALH,CADF;AASD,CAnBD","sourcesContent":["import { boolean, number, select, text } from '@storybook/addon-knobs'\nimport { storiesOf } from '@storybook/react'\nimport * as React from 'react'\nimport { saiyanTheme } from '@/styles/sc-vars-saiyan'\nimport { ThemeColors } from '@/types/theme'\nimport { InfoText } from '..'\nimport { TypographyScaleInfo } from './InfoText.interface'\n\nstoriesOf('Design System/Atoms/Typography', module).add('InfoText', () => {\n  const sampleText = text('Example text', `Lorem ipsum dolor`)\n  const scaleOptions = ['small', 'small-highlighted', 'extra-small']\n  const scale = select('Type scale', scaleOptions, 'small') as TypographyScaleInfo\n  const tagOptions: (keyof JSX.IntrinsicElements)[] = ['span', 'em', 'strong', 'div']\n  const tagName = select('Tag name', tagOptions, 'span')\n  const showColors = boolean('Enable Custom Color', false)\n  const selectedColor = select('Color', Object.keys(saiyanTheme.color), 'black') as ThemeColors\n  const limitLines = number('Limit Lines', 0)\n\n  return (\n    <InfoText\n      tag={tagName}\n      scale={scale}\n      {...(showColors && { color: selectedColor })}\n      limitLines={limitLines}>\n      {sampleText}\n    </InfoText>\n  )\n})\n"],"file":"InfoText.story.js"}