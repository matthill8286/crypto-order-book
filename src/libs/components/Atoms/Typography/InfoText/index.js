export * from "./InfoText";
export * from "./InfoText.interface";
//# sourceMappingURL=index.js.map