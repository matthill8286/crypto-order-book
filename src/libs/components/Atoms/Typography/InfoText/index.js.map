{"version":3,"sources":["../../../../../src/components/Atoms/Typography/InfoText/index.ts"],"names":[],"mappings":"AAAA;AACA","sourcesContent":["export * from './InfoText'\nexport * from './InfoText.interface'\n"],"file":"index.js"}