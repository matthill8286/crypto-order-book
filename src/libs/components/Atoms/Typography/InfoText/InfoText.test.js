import React from 'react';
import { renderWithTheme } from "../../../../testRenderer";
import { InfoText } from "./InfoText";
describe('InfoText', function () {
  it('renders correctly', function () {
    var tree = renderWithTheme(React.createElement(InfoText, {
      scale: "small"
    }));
    expect(tree).toMatchSnapshot();
  });
});
//# sourceMappingURL=InfoText.test.js.map