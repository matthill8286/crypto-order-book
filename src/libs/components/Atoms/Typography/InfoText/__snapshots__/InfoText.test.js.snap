// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoText renders correctly 1`] = `
<span
  className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 fxZwjU"
  color="grey4"
  fontSize="xxs"
/>
`;
